sdk/vppagent/flush.go

```diff
@@ -0,0 +1,172 @@
+package vppagent
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc"
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/pkg/tools"
+	"github.com/networkservicemesh/networkservicemesh/sdk/common"
+	"github.com/networkservicemesh/networkservicemesh/sdk/endpoint"
+	"github.com/opentracing/opentracing-go"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+)
+
+// Flush is a VPP Agent Flush composite
+type Flush struct {
+	endpoint.BaseCompositeEndpoint
+	Conn     *grpc.ClientConn
+}
+
+// Request implements the request handler
+func (f *Flush) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+
+	if f.GetNext() == nil {
+		logrus.Fatal("The VPP Agent Flush composite requires that there is Next set")
+	}
+
+	incomingConnection, err := f.GetNext().Request(ctx, request)
+	if err != nil {
+		logrus.Errorf("Next request failed: %v", err)
+		return nil, err
+	}
+
+	opaque := f.GetNext().GetOpaque(incomingConnection)
+	if opaque == nil {
+		err = fmt.Errorf("received empty data from Next")
+		logrus.Errorf("Unable to find the DataChange: %v", err)
+		return nil, err
+	}
+	dataChange := opaque.(*ConnectionData).DataChange
```

Comment:
You should have a proper check for this type conversion:

https://tour.golang.org/methods/15

also check for and return an error if DataChange is nil.

Reply:
Fixed